### ISSUE TITLE
About performance?

### ISSUE THREAD

**Issue by ShiguangSun** [zchuning/repo#3]
Hello, I ran the distraction DMC experiment. The hyperparameters were set to Table 2 in Appendix C, but the performance was much worse than what was shown in the paper, and the variance was large. Why is this?  Are there other settings that need attention? 
![Cartpole_swingup](https://github.com/zchuning/repo/assets/77392434/aa5d714d-7beb-4e65-81f5-bcc18c331d62)
![hopper_stand](https://github.com/zchuning/repo/assets/77392434/0c876488-8c86-4792-9ced-1a4c8d8e963c)
![cheetah_run](https://github.com/zchuning/repo/assets/77392434/e59cc68d-2768-4722-9677-640207498670)
![wlker_run](https://github.com/zchuning/repo/assets/77392434/ed8675a4-6b61-42b1-aa10-68633f75b461)
![walker_walk](https://github.com/zchuning/repo/assets/77392434/52e06092-495c-4264-aa2b-8cc01d1c1b6d)
![walker_stand](https://github.com/zchuning/repo/assets/77392434/ad784dff-03c7-42fd-9cc7-f3d6b1186d61)

**Comment by zchuning** [comment]
Thank you for your interest in our work! 

Regarding variance: the shadowed regions in the paper's plots represent [standard error of the mean](https://en.wikipedia.org/wiki/Standard_error), which is the standard deviation divided by the number of seeds. You can toggle the same visualization in wandb by editing the panel to display the standard error. 

From the plots it seems you were able to reproduce all the experiments except cartpole swingup and perhaps walker run (cheetah run is even better than the reported results). For walker run, try setting init_beta=5e-6 and that should bump the performance slightly. I'm not quite sure why the carpole performance is so suboptimal, as I'm able to reproduce the paper results using the hyperparameters in Table 2. I would suggest checking if the hyperparameters are set correctly.
<img width="520" alt="Screenshot 2023-12-08 at 3 27 06 PM" src="https://github.com/zchuning/repo/assets/40848789/d07065f7-21f9-42cf-82f5-cf9ed2f412b5">

**Comment by ShiguangSun** [comment]
Thanks for the answer! I will try as you suggest. Except for cartpole and walker run, the convergence performance is similar, but I found that the experimental results I ran converged slower than those shown in the paper. Do you have any suggestions on this?

**Comment by zchuning** [comment]
It's hard to compare your learning curves with the ones in the paper without plotting them together, but it looks like they are converging at roughly the same rate as measured by the number of environment steps. I wouldn't say there is a discrepancy in terms of convergence speed.

To make the x-axis aligned with the paper's plots, you can plot the return with respect to `train/step` in wandb.

**Comment by ShiguangSun** [comment]
ok，thanks！

**Comment by gunnxx** [comment]
Hi, do you have logs for the `log_beta` values? I am wondering how large is the final value of beta and does it differ significantly between environment. I am trying to play around and using a fixed beta.

Anyway, thanks for the nice code

**Comment by zchuning** [comment]
Hi, attached is the training logs of repo on the Distracted DMC environments. The final beta values do vary across environments. I experimented with fixing beta to the converged values, but that resulted in performance drop, which led me to believe that an adaptive beta is important.
[repo_log.zip](https://github.com/zchuning/repo/files/15030120/repo_log.zip)

**Comment by gunnxx** [comment]
Whoaa nice, thanks a lot!
